### PR TITLE
feat: implement the frontEnd for the CommentFunctionnality

### LINKED ISSUE #233: implement the UI for the comment section

### PR DESCRIPTION
**Pull request by ismailIBNMAJAH**
### Overview  
This PR introduces a significant update to the `VideoScreen` component by adding a fully functional comment system. Unit and UI tests have been added to ensure the robustness of the comment system and the overall user experience. Additionally, the codebase has been optimized with local storage management to reduce dependency on Firebase for frequent fetch operations.  

#### Implementation Details  

- **Comment System**:  
  - Added a `Comment Menu` to the `VideoScreen`, enabling users to:  
    - Post new comments.  
    - Like comments.  
    - Delete their own comments.  
  - Seamlessly integrated the comment system with the video experience, offering a dynamic and interactive UI.  

- **Optimized Data Flow**:  
  - Implemented local storage management to minimize redundant fetches from Firebase.  
  - Improved performance and responsiveness by caching data locally, ensuring actions like posting, liking, and deleting comments are handled efficiently.  

#### UI Layer  

- **Comment Menu Implementation**:  
  - Integrated a bottom sheet for the comment menu with dynamic UI states (`Loading`, `Success`, and `Error`).  
  - Used a `LazyColumn` to display comments in an efficient, scrollable list.  
  - Added interactivity with buttons for liking and deleting comments, shown conditionally based on the current user.  

- **Backend Enhancements**:  
  - Introduced dedicated fake classes (`FakeCommentsViewModel`, `FakeCommentsRepository`, etc.) to facilitate testing.(modifications made following feedback from M2.)  
  - Optimized backend operations with a focus on local storage integration to reduce Firebase dependency.  

#### Unit and UI Tests  

- **Unit Tests**:  
  - Tested the `CommentsRepository` and `CommentsViewModel` for robust data handling and UI state updates.  
  - Verified correct behavior for adding, liking, and deleting comments.  

- **UI Tests**:  
  - Created comprehensive tests for the `VideoScreen` using `ComposeTestRule`:  
    - Verified the visibility and functionality of the comment menu.  
    - Tested interactions for posting, liking, and deleting comments.  
    - Ensured error handling and edge cases are gracefully managed.  

- **Fake Classes**:  
  - Developed multiple fake classes to mock dependencies, streamlining the testing process and enabling future test scalability.  

### User Experience Improvements  
- **Dynamic Comment Interaction**:  
  - Users can interact with comments in real-time, enhancing the social aspect of the `VideoScreen`.  
  - Clear visual feedback for actions like liking and deleting.  

### Related Issues  
Closes #233: Add a comment system to the `VideoScreen` and implement related tests.  

### Screenshots  
comment1  

comment2

### CODE REVIEW

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt [hunk 1]
```diff
@@ -39,134 +55,372 @@ import com.github.se.eduverse.ui.navigation.BottomNavigationMenu
...
+fun CommentsMenuContent(
+    publicationId: String,
+    commentsViewModel: CommentsViewModel,
+    currentUserId: String,
+    onDismiss: () -> Unit
+) {
+  Box(modifier = Modifier.fillMaxWidth().background(Color.White).testTag("CommentsMenuContent")) {
+    CommentSection(publicationId, commentsViewModel, currentUserId)
+  }
+}
```
Comment (MathieuxC) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:260]: You don't use the onDismiss argument here

File: app/src/main/java/com/github/se/eduverse/viewmodel/PublicationViewModel.kt [hunk 2]
```diff
@@ -22,6 +24,8 @@ open class PublicationViewModel(private val repository: PublicationRepository) :
   open val error: StateFlow<String?>
     get() = _error
 
+  private val userProfileCache = mutableMapOf<String, String?>()
```
Comment (MathieuxC) [inline app/src/main/java/com/github/se/eduverse/viewmodel/PublicationViewModel.kt:27]: You added a cache but never use it

**Review by MathieuxC — APPROVED** [review]
Very good PR ! The feature is good and I didn't notice any issue with your code. There are a couple places where you don't follow conventions (see comments) but otherwise this is very impressive.
Also the PR is very big, in the future try to make it in smaller chunk if possible

File: app/src/androidTest/java/com/github/se/eduverse/fake/fakeCommentsRepository.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.github.se.eduverse.fake
+
```
[the remark this answers is not in the document]
Comment (MathieuxC) [reply app/src/androidTest/java/com/github/se/eduverse/fake/fakeCommentsRepository.kt:2]: fakeCommentsRepository.kt -> FakeCommentsRepository.kt

File: app/src/androidTest/java/com/github/se/eduverse/fake/FakeCommentsViewModel.kt
Comment (mehdixboutaleb) [inline app/src/androidTest/java/com/github/se/eduverse/fake/FakeCommentsViewModel.kt:1]: great class for the testing instead of mocking, lgtm

File: app/src/androidTest/java/com/github/se/eduverse/fake/FakeNavigationActions.kt
Comment (mehdixboutaleb) [inline app/src/androidTest/java/com/github/se/eduverse/fake/FakeNavigationActions.kt:1]: nice implementation will be useful for the next test that will be written

File: app/src/androidTest/java/com/github/se/eduverse/fake/FakeProfileRepository.kt
Comment (mehdixboutaleb) [inline app/src/androidTest/java/com/github/se/eduverse/fake/FakeProfileRepository.kt:1]: lgtm

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt
Comment (mehdixboutaleb) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:1]: lgtm seems complete and well implemented well done

**Review by mehdixboutaleb — CHANGES_REQUESTED** [review]
This new feature is a great advancement in the project and showed your potential, make sure to change the comments that are in French to English for the consistency of the project apart from that everything looks pretty good
